### PR TITLE
Check empty file uploads, tests for the same and tests for local uploads on file interceptor

### PR DESCRIPTION
**Pull request by gnaaruag**
## Checklist
Make sure you have
- [ ] Added relevant `@nestjs/swagger` decorators wherever required 
- [x] Added test cases (unit and integration) wherever required

### Tests in #50 
This PR adds tests for local uploads described in #50 to `file-upload.interceptor.spec.ts`
also adds e2e tests in the file upload sample app in `sample/04-file-upload/test/app.e2e-spec.ts`

### fix for issue #156 
- this adds a empty file upload check to the `file-upload.controller.ts` in `packages/common/src` 
- also adds tests for correct file upload, empty file upload
- if the file is found to be empty a status code of 400 is sent, else it is uploaded further according to `process.env.STORAGE_MODE`
- in `file-upload.interceptor` the line `request.raw = request.raw || request;` allows for valid request body to pass through

## Steps to test #50 
Navigate to `packages/common/test` and run `npm run test interceptor/file-upload.interceptor.spec.ts`
Navigate to `sample/04-file-upload` and run `npm run test:e2e`

## Steps to test #156 

**to test manually** 
create a post request to `/file/upload-file`  
test with empty file and non empty file via `multipart/form-data`

**to run automated test**
navigate to `samples/04-file-upload` and run `npm run test:e2e`

### CODE REVIEW

File: packages/common/test/interceptor/file-upload.interceptor.spec.ts [hunk 1]
```diff
@@ -59,8 +104,29 @@ describe('FastifyFileInterceptor', () => {
     expect(nextHandler.handle).toHaveBeenCalled();
   });
 
+  it('should throw error when process.env.STORAGE_ENDPOINT is not defined or empty', async () => {
+    // mocking env vars
+    const OLD_ENV = process.env;
+    
+  });
```
Comment (techsavvyash) [inline packages/common/test/interceptor/file-upload.interceptor.spec.ts]: is this expected.

Comment (gnaaruag) [reply packages/common/test/interceptor/file-upload.interceptor.spec.ts]: seems to have been a mix up, will fix it asap

**Review by techsavvyash — APPROVED** [review]

**Comment by techsavvyash** [comment]
Merging this thanks for the effort @gnaaruag

**Comment by VedantKhairnar** [comment]
Hello @techsavvyash 
Will you pls close the corresponding issues as well if the required PR is merged?
Thanks.

**Comment by VedantKhairnar** [comment]
Hey there,
Thanks for your contribution.
As part of Augtoberfest, we want to reward you for your contributions, so pls fill out [this form](https://forms.gle/VBDUPWMYcZEbmVrQ9) by 19th Sept.
Thanks.
